tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -165,79 +189,127 @@ struct bt_mesh_cfg_cli_mod_pub pub_params = {
 
-	k_sleep(K_SECONDS(SUSPEND_DURATION));
+	bt_mesh_test_parse_mesh_gatt_preamble(buf);
 
-	ASSERT_OK(bt_mesh_resume());
-	suspended = false;
-	LOG_INF("Mesh resumed.");
+	if (gatt_param.service == MESH_SERVICE_PROVISIONING) {
+		LOG_DBG("Parsing pb_gatt adv");
+		bt_mesh_test_parse_mesh_pb_gatt_service(buf);
+	} else {
+		LOG_DBG("Parsing proxy adv");
+		bt_mesh_test_parse_mesh_proxy_service(buf);
+	}
 
-	for (int i = 0; i < NUM_PUB; i++) {
-		ASSERT_OK_MSG(k_sem_take(&publish_sem, K_SECONDS(30)), "Pub timed out");
+	if (gatt_check_rx_count(gatt_param.transmits)) {
+		LOG_DBG("rx completed, stopping scan...");
+		k_sem_give(&publish_sem);
 	}
+}
 
-	/* Allow publishing to finish before suspending. */
-	k_sleep(K_MSEC(100));
-	ASSERT_OK(bt_mesh_suspend());
+static void gatt_rx_beacons(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type       = BT_HCI_LE_SCAN_PASSIVE,
+		.options    = BT_LE_SCAN_OPT_NONE,
+		.interval   = BT_MESH_ADV_SCAN_UNIT(1000),
+		.window     = BT_MESH_ADV_SCAN_UNIT(1000)
+	};
+	int err;
 
-	PASS();
+	err = bt_le_scan_start(&scan_param, gatt_scan_cb);
+	ASSERT_FALSE_MSG(err && err != -EALREADY, "Starting scan failed (err %d)\n", err);
+
+	err = k_sem_take(&publish_sem, K_SECONDS(10));
+	if (err && (dut_status == DUT_RUNNING)) {
+		FAIL("Tester timed out");
+	} else if ((err == -EAGAIN) && (dut_status == DUT_SUSPENDED)) {
+		LOG_WRN("Tester timed out (DUT is suspended)");
+	}
+
+	err = bt_le_scan_stop();
+	ASSERT_FALSE_MSG(err && err != -EALREADY, "Stopping scan failed (err %d)\n", err);
 }
 
-static void test_rx_suspend_resume(void)
+static void suspend_state_change_cb(const bt_addr_le_t *addr, int8_t rssi, uint8_t adv_type,
+			struct net_buf_simple *buf)
 {
-	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
-	bt_mesh_device_setup(&prov, &rx_comp);
-	provision_and_configure(rx_cfg, 1);
-	k_sem_init(&publish_sem, 0, 1);
+	uint8_t length;
 
-	/* Receive messages before and after suspending. A publication may get lost
-	 * when suspending immeditiately after publication, thus the "-1".
-	 */
-	for (int i = 0; i < NUM_PUB * 2 - 1; i++) {
-		ASSERT_OK_MSG(k_sem_take(&publish_sem, K_SECONDS(30)), "Receiver timed out");
+	if (adv_type != BT_GAP_ADV_TYPE_ADV_NONCONN_IND) {
+		return;
 	}
 
-	PASS();
+	length = net_buf_simple_pull_u8(buf);
+	ASSERT_EQUAL(buf->len, length);
+	ASSERT_EQUAL(length, sizeof(uint8_t) + sizeof(enum dut_mesh_status));
+	ASSERT_EQUAL(BT_DATA_MESH_MESSAGE, net_buf_simple_pull_u8(buf));
+
+	enum dut_mesh_status *msg_status =
+		net_buf_simple_pull_mem(buf, sizeof(enum dut_mesh_status));
+
+	if ((*msg_status == DUT_RUNNING) || (*msg_status == DUT_SUSPENDED)) {
+		dut_status = *msg_status;
+	} else {
+		FAIL("Received unexpected data");
+	}
+
+	LOG_DBG("Received %d from DUT, setting status to %s",
+		*msg_status, (dut_status == DUT_SUSPENDED) ? "true" : "false");
+	k_sem_give(&publish_sem);
 }
 
-static void test_tx_suspend_disable_resume(void)
+static void dut_pub_body(bool disable_bt)
```

Comment:
```suggestion
static void dut_pub_common(bool disable_bt)
```

Reply:
Changed to your suggestion, same for below.